discirc/irc_bot.py

```diff
@@ -70,7 +71,10 @@ def on_connect(self):
         if self.password:
             self.send('PASS', password=self.password)
         for chan in self.channels.values():
-            self.send('JOIN', channel=chan)
+            if chan not in self.channelPass.values():
```

Comment:
Since you invert keys and values in the config it should be 

`if chan not in self.channelPass`